src/HttpUserAgentParser/HttpUserAgentParser.cs

```diff
@@ -206,45 +209,133 @@ private static bool TryExtractVersion(ReadOnlySpan<char> haystack, out Range ran
     {
         range = default;
 
-        // Limit search window to avoid scanning entire UA string unnecessarily
-        const int Window = 128;
-        if (haystack.Length > Window)
-        {
-            haystack = haystack.Slice(0, Window);
-        }
+        // Vectorization is used in a optimistic way and specialized to common (trimmed down) user agents.
+        // When the first two char-vectors don't yield any success, we fall back to the scalar path.
+        // This penalized not found versions, but has an advantage for found versions.
+        // Vector512 is left out, because there are no common inputs with length 128 or more.
+        //
+        // Two short (same size as char) vectors are read, then packed to byte vectors on which the
+        // operation is done. For short / chart the higher byte is not of interest and zero or outside
+        // the target characters, thus with bytes we can process twice as much elements at once.
 
-        // Find first digit
-        int start = -1;
-        for (int i = 0; i < haystack.Length; i++)
+        if (Vector256.IsHardwareAccelerated && haystack.Length >= 2 * Vector256<short>.Count)
```

Comment:
I'm pretty sure the implementation is correct, but should we add some kind of toggle (env variable) with which vectorization can be disabled in case there's a bug in the code (i.e. from some strange user agent that we don't know at the moment), so a user could still use the lib w/o the need to wait for hotfix release?

Reply:
> Further the biggest speed-up may come from a not linear scan through all possible patterns.

In my head I'm ready with such an approach, but I need to finish a work-project, then I'll prototype it and see how it goes.

Reply:
Feel free, have complete confidence in your everything looks impressively good!

Reply:
The first try with that idea is nice, as the user agent only needs to be scaned once ($O(n)$) and not for every possibility.
But for some browsers, where we depend on the order in the arrays, the wrong result is yielded. So I have to research a bit more. 

Maybe there will move also some things around (w/o API breaking changes), so I'd like to leave this PR open so that all together can be done or only parts of it.

I hope to continue on this next weeks.